### PR TITLE
Console - Migrate endpoints to mapiv2

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://gravitee.atlassian.net/browse/APIM-2795

## Description

Finish migrating V2 endpoints screens and use MAPI v2

## Additional context

localhost_3000_

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pdloyfydtt.chromatic.com)

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://pr.team-apim.gravitee.dev/5655/console](https://pr.team-apim.gravitee.dev/5655/console)
      Portal: [https://pr.team-apim.gravitee.dev/5655/portal](https://pr.team-apim.gravitee.dev/5655/portal)
      Management-api: [https://pr.team-apim.gravitee.dev/5655/api/management](https://pr.team-apim.gravitee.dev/5655/api/management)
      Gateway v4: [https://pr.team-apim.gravitee.dev/5655](https://pr.team-apim.gravitee.dev/5655)
      Gateway v3: [https://pr.gateway-v3.team-apim.gravitee.dev/5655](https://pr.gateway-v3.team-apim.gravitee.dev/5655)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-atnrysginc.chromatic.com)

### CODE REVIEW

**Comment by ThibaudAV** [comment]
I'll accept duplication in keyStore and turstStrore. They're not exactly the same component, so it's good if they can have a life of their own.

**Comment by gaetanmaisse** [comment]
It looks ok except the Key password is not saved 🤔

**Comment by ThibaudAV** [comment]
Good point. We've mixed V4 and V2 apis here when it's only V2. 
I've added one commit to align with the V2 api definition.
And another one to only keep openapi to validate api endpoint config payload. But this means that MapiV1 no longer validates this part. 🤷‍♂️ I don't think it's a problem, our console uses mapiV2 and even for a client it's better to use Mapi v2 for a 360° experience.

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/endpoint-http-config/endpoint-http-config.component.ts [hunk 1]
```diff
@@ -225,20 +243,31 @@ export class EndpointHttpConfigComponent implements OnInit, OnDestroy {
...
+          httpProxy.get('host').disable({ onlySelf: true });
+          httpProxy.get('port').clearValidators();
+          httpProxy.get('port').disable({ onlySelf: true });
+          httpProxy.get('type').disable({ emitEvent: false });
+          httpProxy.get('username').disable({ emitEvent: false });
+          httpProxy.get('password').disable({ emitEvent: false });
         }
+
+        // Update validators
+        Object.keys(httpProxy.controls).forEach((controlName) => {
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/endpoint-http-config/endpoint-http-config.component.ts]: I would have guess that `httpProxy.updateValueAndValidity({ emitEvent: false })` would have do the same than looping on form keys no ?
And I don't really get why after that you still need to call the updateValueAndValidity but send an event this time ?

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/endpoint-http-config/endpoint-http-config.component.ts [hunk 2]
```diff
@@ -225,20 +243,31 @@ export class EndpointHttpConfigComponent implements OnInit, OnDestroy {
...
-          httpProxy.get('username').enable();
-          httpProxy.get('password').enable();
+          httpProxy.get('useSystemProxy').enable({ onlySelf: true });
+          httpProxy.get('host').addValidators(Validators.required);
+          httpProxy.get('host').enable({ onlySelf: true });
+          httpProxy.get('port').addValidators(Validators.required);
+          httpProxy.get('port').enable({ onlySelf: true });
+          httpProxy.get('type').enable({ emitEvent: false });
+          httpProxy.get('username').enable({ emitEvent: false });
+          httpProxy.get('password').enable({ emitEvent: false });
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/endpoint-http-config/endpoint-http-config.component.ts]: I don't know if it's a good idea and you can let it like that but it looks like some part of the code could be mutualised between this valueChanges and the one from useSystemProxy no ? 🤷🏼

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          provide: ControlContainer,
+          useValue: {},
+        },
+      ],
+    })
+      .overrideProvider(InteractivityChecker, {
+        useValue: {
+          isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        },
+      })
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.spec.ts]: I think we can remove it

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+      });
+  }
+
+  ngDoCheck() {
+    const parentFormControl = this.controlContainer?.control?.get(this.formControlName);
+    if (parentFormControl) {
+      // If parent form control is touched, mark all internal fields as touched
+      if (parentFormControl.touched && !this.parentIsTouched) {
+        this.internalFormGroup.markAllAsTouched();
+        // this.internalFormGroup.updateValueAndValidity({ emitEvent: false });
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.ts]:
```suggestion

```

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+          invalidKeyStore: true,
+        };
+  }
+
+  writeValue(value: KeyStore): void {
+    this.internalFormGroup.reset(keyStoreTypeToInternalFormValue(value));
+  }
+}
+
+const pathOrContentRequired: (pathControlName: string, contentControlName: string) => ValidatorFn = (
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/ssl-keystore-form/ssl-keystore-form.component.ts:221]: do you think that we could have specific small unit tests for the validators ?

**Review by Okhelifi — APPROVED** [review]

**Pull request merged**